Source/Core/VideoBackends/OGL/TextureCache.cpp

```diff
@@ -351,18 +370,158 @@ void TextureCache::CompileShaders()
+			b = Convert5To8((val) & 0x1f);
+			a = 0xFF;
+			return float4(r, g, b, a) / 255;
+		}
+
+		float4 DecodePixel_IA8(int val)
+		{
+			int i = val & 0xFF;
+			int a = val >> 8;
+			return float4(i, i, i, a) / 255;
+		}
+
+		void main()
+		{
+			int src = int(round(texture(samp9, f_uv0).r * multiplier));
+			src = int(texelFetch(samp10, src + texture_buffer_offset).r);
+			src = ((src << 8) & 0xFF00) | (src >> 8);
+			ocol0 = DECODE(src);
+		}
+		)GLSL";
+
+	if (g_Config.backend_info.bSupportsPaletteConversion)
+	{
+		ProgramShaderCache::CompileShader(
+			s_palette_pixel_shader[GX_TL_IA8],
+			StringFromFormat(VProgram, prefix, prefix).c_str(),
+			("#define DECODE DecodePixel_IA8" + palette_shader).c_str(),
+			GProgram);
+		s_palette_buffer_offset_uniform[GX_TL_IA8] = glGetUniformLocation(s_palette_pixel_shader[GX_TL_IA8].glprogid, "texture_buffer_offset");
+		s_palette_multiplier_uniform[GX_TL_IA8] = glGetUniformLocation(s_palette_pixel_shader[GX_TL_IA8].glprogid, "multiplier");
+		s_palette_copy_position_uniform[GX_TL_IA8] = glGetUniformLocation(s_palette_pixel_shader[GX_TL_IA8].glprogid, "copy_position");
+
+		ProgramShaderCache::CompileShader(
+			s_palette_pixel_shader[GX_TL_RGB565],
+			StringFromFormat(VProgram, prefix, prefix).c_str(),
+			("#define DECODE DecodePixel_RGB565" + palette_shader).c_str(),
+			GProgram);
+		s_palette_buffer_offset_uniform[GX_TL_RGB565] = glGetUniformLocation(s_palette_pixel_shader[GX_TL_RGB565].glprogid, "texture_buffer_offset");
+		s_palette_multiplier_uniform[GX_TL_RGB565] = glGetUniformLocation(s_palette_pixel_shader[GX_TL_RGB565].glprogid, "multiplier");
+		s_palette_copy_position_uniform[GX_TL_RGB565] = glGetUniformLocation(s_palette_pixel_shader[GX_TL_RGB565].glprogid, "copy_position");
+
+		ProgramShaderCache::CompileShader(
+			s_palette_pixel_shader[GX_TL_RGB5A3],
+			StringFromFormat(VProgram, prefix, prefix).c_str(),
+			("#define DECODE DecodePixel_RGB5A3" + palette_shader).c_str(),
+			GProgram);
+		s_palette_buffer_offset_uniform[GX_TL_RGB5A3] = glGetUniformLocation(s_palette_pixel_shader[GX_TL_RGB5A3].glprogid, "texture_buffer_offset");
+		s_palette_multiplier_uniform[GX_TL_RGB5A3] = glGetUniformLocation(s_palette_pixel_shader[GX_TL_RGB5A3].glprogid, "multiplier");
+		s_palette_copy_position_uniform[GX_TL_RGB5A3] = glGetUniformLocation(s_palette_pixel_shader[GX_TL_RGB5A3].glprogid, "copy_position");
+	}
 }
 
 void TextureCache::DeleteShaders()
 {
 	s_ColorMatrixProgram.Destroy();
 	s_DepthMatrixProgram.Destroy();
+
+	if (g_Config.backend_info.bSupportsPaletteConversion)
+		for (auto& shader : s_palette_pixel_shader)
+			shader.Destroy();
 }
 
-void TextureCache::ConvertTexture(TextureCache::TCacheEntryBase* entry, TCacheEntryBase* unconverted, void* palette, TlutFormat format)
+void TextureCache::ConvertTexture(TCacheEntryBase* _entry, TCacheEntryBase* _unconverted, void* palette, TlutFormat format)
 {
-	// TODO: Implement.
-	return;
+	g_renderer->ResetAPIState();
```

Comment:
Just double checked, TextureCacheBase still calls this function at all times, so we need a check inside this function for if we support palette conversion